src/sysc/kernel/sc_module.h

```diff
@@ -400,6 +400,10 @@ extern SC_API sc_module* sc_module_dynalloc(sc_module*);
     typedef user_module_name SC_CURRENT_USER_MODULE;                          \
     user_module_name( ::sc_core::sc_module_name )
 
+#define SC_DTOR(user_module_name)                                             \
+    typedef user_module_name SC_CURRENT_USER_MODULE;                          \
+    ~##user_module_name()
+
```

Comment:
This is not part of IEEE Std 1666-2011 and adding new preprocessor macros is IMHO not good coding practice. Personally, I prefer to avoid its counterpart `SC_CTOR` as well and use the regular C++ constructor syntax. This makes the code more clear for the readed and tools.